packages/sitecore-jss/src/site/graphql-redirects-service.ts

```diff
@@ -80,11 +80,12 @@ export class GraphQLRedirectsService {
 
   /**
    * Fetch an array of redirects from API
+   * @param {string} [dynamicSiteName] dynamic site name
    * @returns Promise<RedirectInfo[]>
    * @throws {Error} if the siteName is empty.
    */
-  async fetchRedirects(): Promise<RedirectInfo[]> {
-    const siteName: string = this.options.siteName;
+  async fetchRedirects(dynamicSiteName?: string): Promise<RedirectInfo[]> {
```

Comment:
I'm wondering if we should make use of the `SiteResolver` in the personalize and redirects middlewares as well. That would mean we can get rid of `siteName` configs as well. e.g.

1. Remove `siteName` option on middleware config, and replace with `getSite` function
2. In middleware, use `sc_site` if present else this to lookup by hostname
3. Use `siteResolver.getByHost` in plugins
4. Make `siteName` a required parameter in `GraphQLRedirects/Personalize` fetch/get methods
5. Remove `siteName` from  `GraphQLRedirects/Personalize` configs

Reply:
Great idea! Makes sense for me, applied